### ISSUE TITLE
Transform functions device placement errors (pytorch)

### ISSUE THREAD

**Issue by jjennings955** [BachiLi/redner#99]
The transform functions such as [gen_translate_matrix](https://github.com/BachiLi/redner/blob/3f7c0018e9e1accccd754446477169111a75eaa4/pyredner/transform.py#L26) don't function when using cuda.

The error is:
```
  File "/home/jason/miniconda2/envs/f/lib/python3.7/site-packages/pyredner/transform.py", line 29, in gen_translate_matrix
    return torch.stack([torch.cat([o, z, z, translate[0:1]], 0),
RuntimeError: Expected object of backend CUDA but got backend CPU for sequence element 0 in sequence argument at position #1 'tensors'
```

Intermediate tensors they create are placed on CPU.

[gen_rotate_matrix](https://github.com/BachiLi/redner/blob/3f7c0018e9e1accccd754446477169111a75eaa4/pyredner/transform.py#L44) handles it appropriately, by doing `rot_x = torch.zeros((3, 3), device=angles.device, dtype=torch.float32)`

Also, it would be nice if these functions worked on batches :)

**Comment by BachiLi** [comment]
Fixed in https://github.com/BachiLi/redner/commit/cb1d3e34e71796079d55944aa90784a7888c46ca
To be honest these are just some convenient utility functions I wrote for the papers and examples. I didn't really want to spend too much time maintaining them. I welcome any contribution to the batching mechanism.

Also for the sake of performance, it is probably faster to generate these matrices on CPUs and then copy them to GPUs. Even so, PyTorch and TensorFlow are really bad at these small granularity operations.

**Comment by tetterl** [comment]
Copy&paste error fixed in https://github.com/BachiLi/redner/pull/100

**Comment by BachiLi** [comment]
Thanks!

**Comment by jjennings955** [comment]
> Fixed in [cb1d3e3](https://github.com/BachiLi/redner/commit/cb1d3e34e71796079d55944aa90784a7888c46ca)
> To be honest these are just some convenient utility functions I wrote for the papers and examples. I didn't really want to spend too much time maintaining them. I welcome any contribution to the batching mechanism.
> 
> Also for the sake of performance, it is probably faster to generate these matrices on CPUs and then copy them to GPUs. Even so, PyTorch and TensorFlow are really bad at these small granularity operations. Ideally you want to do these in C++.



> Copy&paste error fixed in #100

Thanks to you both!
